templates/docker-compose/CeleryExecutor/docker-compose.yml

```diff
@@ -0,0 +1,97 @@
+version: '3'
+services:
+  postgres:
+    image: postgres:latest
+    container_name: postgres_cont
+    environment:
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=postgres
+      - POSTGRES_DB=airflow
+      - POSTGRES_PORT=5432
+    ports:
+      - 5432:5432
+
+  redis:
+    image: redis:latest
+    container_name: redis_cont
+    restart: always
+    ports:
+      - 6379:6379
+
+  airflow:
+    image: apache/airflow:1.10.10
+    container_name: airflow_cont
+    env_file:
+      - airflow-env-variables.env
+    restart: always
+    command:  webserver
+    ports:
+      - 8080:8080
+    volumes:
+      - ./airflow-data/dags:/opt/airflow/dags
+      - ./airflow-data/logs:/opt/airflow/logs
+      - ./airflow-data/plugins:/opt/airflow/plugins
+
+  airflow-scheduler:
+    image: apache/airflow:1.10.10
+    container_name: airflow_scheduler_cont
+    env_file:
+      - airflow-env-variables.env
+    restart: always
+    command: scheduler
+    volumes:
+      - ./airflow-data/dags:/opt/airflow/dags
+      - ./airflow-data/logs:/opt/airflow/logs
+      - ./airflow-data/plugins:/opt/airflow/plugins
+
+  airflow-worker1:
+    image: apache/airflow:1.10.10
+    container_name: airflow_worker1_cont
+    env_file:
+      - airflow-env-variables.env
+    restart: always
+    command: worker
+    volumes:
+      - ./airflow-data/dags:/opt/airflow/dags
+      - ./airflow-data/logs:/opt/airflow/logs
+      - ./airflow-data/plugins:/opt/airflow/plugins
+
+  airflow-worker2:
+    image: apache/airflow:1.10.10
+    container_name: airflow_worker2_cont
+    env_file:
+      - airflow-env-variables.env
+    restart: always
+    command: worker
+    volumes:
+      - ./airflow-data/dags:/opt/airflow/dags
+      - ./airflow-data/logs:/opt/airflow/logs
+      - ./airflow-data/plugins:/opt/airflow/plugins
+
+
+  airflow-worker3:
+    image: apache/airflow:1.10.10
+    container_name: airflow_worker3_cont
+    env_file:
+      - airflow-env-variables.env
+    command: worker
+    restart: always
+    volumes:
+      - ./airflow-data/dags:/opt/airflow/dags
+      - ./airflow-data/logs:/opt/airflow/logs
+      - ./airflow-data/plugins:/opt/airflow/plugins
+
+  airflow-flower:
+    image: apache/airflow:1.10.10
+    container_name: airflow_flower_cont
+    restart: always
+    volumes:
```

Comment:
Can you use [extension-fields]( https://docs.docker.com/compose/compose-file/#extension-fields) to reduce code duplication?
https://medium.com/@kinghuang/docker-compose-anchors-aliases-extensions-a1e4105d70bd#f191

Reply:
I did contemplate/consider using extension-fields, however IMO we shouldn't incorporate extension-fields because of following reason for this usecase.
- Most production environments like ECS/Fargate and Docker swarm don't support them unlike docker env variables.
- Users might want to use different images and volumes for different airflow services. (Airflow workers might need java, specific python packages etc.,)
- Much of code duplication is eliminated in form of .env master file already.
- Readability and Adaptation of extension-fields is still lacking with majority of docker users.
-  Container names should be unique.
- Adavanced users can always configure the config according to their needs.

So I believe especially for this usecase extension-fields come under "premature optimization".

_“The real problem is that programmers have spent far too much time worrying about efficiency in the wrong places and at the wrong times; premature optimization is the root of all evil (or at least most of it) in programming.”_ - Donald Knuth